Source/WebCore/rendering/updating/RenderTreeBuilderList.cpp

```diff
@@ -101,12 +102,28 @@ void RenderTreeBuilder::List::updateItemMarker(RenderListItem& listItemRenderer)
     auto newStyle = listItemRenderer.computeMarkerStyle();
     RenderPtr<RenderListMarker> newMarkerRenderer;
     auto* markerRenderer = listItemRenderer.markerRenderer();
-    if (markerRenderer)
+
+    // Rebuild everything if the contents have changed.
+    if (markerRenderer) {
+        auto* oldContentData = markerRenderer->style().contentData();
+        if (oldContentData != newStyle.contentData()) {
+            m_builder.destroy(*markerRenderer);
+            markerRenderer = nullptr;
+        }
+    }
+
+    bool markerHasContent = newStyle.hasContent();
+    if (markerRenderer) {
         markerRenderer->setStyle(WTFMove(newStyle));
+        if (markerHasContent)
+            RenderTreeUpdater::GeneratedContent::updateStyleForContentRenderers(*markerRenderer, markerRenderer->style());
+    }
     else {
         newMarkerRenderer = WebCore::createRenderer<RenderListMarker>(listItemRenderer, WTFMove(newStyle));
         newMarkerRenderer->initializeStyle();
         markerRenderer = newMarkerRenderer.get();
+        if (markerHasContent)
+            RenderTreeUpdater::GeneratedContent::createContentRenderers(m_builder, *markerRenderer, markerRenderer->style(), PseudoId::Marker);
```

Comment:
shouldn't need `RenderTreeUpdater::GeneratedContent::` here as we are in the class.

Reply:
This is `RenderTreeBuilder`, the other class is `RenderTreeUpdater`